text/trace/0173-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,242 @@
+   message was processed. In exceptional cases (fire-and-forget), the
+   settlement stage does not exist.
+
+The messaging semantic conventions need to define how to model those stages in
+traces, how to propagate context, and how to enrich traces with attributes.
+Failures and retries need to be handled in all stages that interface with the
+intermediary (publish, receive and settle) and will be covered by general
+instrumentation guidance.
+
+Based on this model, the following scenarios capture major requirements and
+can be used for prototyping, as examples, and as test cases.
+
+### Individual settlement
+
+Individual settlement systems imply independent logical message flows. A single
+message is created and published in the same context, and it's delivered,
+consumed, and settled as a single entity. Each message needs to be settled
+individually. Usually, settlement information is stored by the intermediary, not
+by the consumer.
+
+Transport batching can be treated as a special case: messages can be
+transported together as an optimization, but are produced and consumed
+individually.
+
+```
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+|Message A| |Message B| |Message C| |Message D| |Message E| |Message F|
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+ Settled                 Settled                             Settled 
+```
+
+#### Examples
+
+1. The following configurations should be instrumented and tested for RabbitMQ
+   or a similar messaging system:
+
+   * 1 producer, 1 queue, 2 consumers
+   * 1 producer, fanout exchange to 2 queues, 2 consumers
+   * 2 producers, fanout exchange to 2 queues, 2 consumers
+
+   Each of the producers continuously produces messages.
+
+### Checkpoint-based settlement
+
+Messages are processed as a stream and settled to specific checkpoints. A
+checkpoint points to a position of the stream up to which messages were
+processed and settled. Messages cannot be settled individually, instead, the
+checkpoint needs to be forwarded. Usually, the consumer is responsible for
+storing checkpointing information, not the intermediary.
+
+Checkpoint-based settlement systems are designed to efficiently receive and
+settle batches of messages. However, it is not possible to settle messages
+independent of their position in the stream (e. g., if message B is located at
+a later position in the stream than message A, then message B cannot be settled
+without also settling message A).
+
+```
+                               Checkpoint
+                                   | 
+                                   v                                
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+|Message A| |Message B| |Message C| |Message D| |Message E| |Message F|
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+                     <---  Settled
```

Comment:
Looking at this figure now the one above is more clear to me, maybe you could give the former some description.